mod/wikirate_source/set/type/source/source.rb

```diff
@@ -34,7 +34,9 @@ def indirect_contributor_search_args
   end
 end
 
-event :process_source_url, :before=>:process_subcards, :on=>:create, :when=>proc{ |c| not (c.subcards["+File"] or c.subcards["+Text"]) } do
+event :process_source_url, :before=>:process_subcards, :on=>:create, :when=>proc{  |c| 
+  not ( (file_card = c.subcards["+File"] and file_card.stringify_keys.has_key?"content")  or ( text_card =  c.subcards["+Text"] and text_card_content = (text_card.stringify_keys)["content"] and !text_card_content.empty? )) 
```

Comment:
https://github.com/bbatsov/ruby-style-guide#no-and-or-or and https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition